ui/apps/platform/src/routePaths.ts

```diff
@@ -102,6 +106,119 @@ export const vulnerabilitiesWorkloadCveDeploymentSinglePath = `${vulnerabilities
+
+type RouteDescription = {
+    featureFlagDependency?: FeatureFlagEnvVar; // can add array as alternative if needed in the future
+    resourceAccessRequirements: ResourceName[]; // assume READ_ACCESS
+};
+
+// Add path variables in alphabetical order to minimize merge conflicts when multiple people add routes.
+const routeDescriptionMap: Record<string, RouteDescription> = {
+    [accessControlPath]: {
+        resourceAccessRequirements: [],
+    },
+    [apidocsPath]: {
+        resourceAccessRequirements: [],
+    },
+    [clustersDelegateScanningPath]: {
+        resourceAccessRequirements: [],
+    },
+    [clustersPathWithParam]: {
+        resourceAccessRequirements: [],
+    },
+    [collectionsPath]: {
+        resourceAccessRequirements: ['WorkflowAdministration'],
+    },
+    [compliancePath]: {
+        resourceAccessRequirements: [],
+    },
+    [complianceEnhancedBasePath]: {
+        resourceAccessRequirements: [],
+    },
+    [configManagementPath]: {
+        resourceAccessRequirements: [],
+    },
+    [dashboardPath]: {
+        resourceAccessRequirements: [],
+    },
+    [integrationsPath]: {
+        resourceAccessRequirements: [],
+    },
+    [listeningEndpointsBasePath]: {
+        resourceAccessRequirements: [],
+    },
+    [networkPath]: {
+        resourceAccessRequirements: [],
+    },
+    [policyManagementBasePath]: {
+        resourceAccessRequirements: [],
+    },
+    [riskPath]: {
+        resourceAccessRequirements: [],
+    },
+    [searchPath]: {
+        resourceAccessRequirements: [],
+    },
+    [systemConfigPath]: {
+        resourceAccessRequirements: [],
+    },
+    [systemHealthPath]: {
+        resourceAccessRequirements: [],
+    },
+    [userBasePath]: {
+        resourceAccessRequirements: [],
+    },
+    [violationsPath]: {
+        resourceAccessRequirements: [],
+    },
+    [vulnManagementPath]: {
+        resourceAccessRequirements: [],
+    },
+    [vulnManagementReportsPath]: {
+        resourceAccessRequirements: ['WorkflowAdministration'],
+    },
+    [vulnManagementRiskAcceptancePath]: {
+        resourceAccessRequirements: [],
+    },
+    [vulnerabilitiesWorkloadCvesPath]: {
+        featureFlagDependency: 'ROX_VULN_MGMT_WORKLOAD_CVES',
+        resourceAccessRequirements: [],
+    },
+    [vulnerabilityReportsPath]: {
+        featureFlagDependency: 'ROX_VULN_MGMT_REPORTING_ENHANCEMENTS',
+        resourceAccessRequirements: ['WorkflowAdministration'],
+    },
+};
+
+type RoutePredicates = {
+    hasReadAccess: HasReadAccess;
+    isFeatureFlagEnabled: IsFeatureFlagEnabled;
+};
+
+export function isRouteEnabled(
+    { hasReadAccess, isFeatureFlagEnabled }: RoutePredicates,
+    path: string
+) {
+    const routeDescription = routeDescriptionMap[path];
+
+    if (!routeDescription) {
+        // eslint-disable-next-line no-console
+        console.warn(`isRouteEnabled for unknown path ${path}`);
+        return true;
```

Comment:
What is the impact of returning `true` here for unknown routes, versus returning `false`?

Reply:
Good question. Return `false` is possible, because the dependency in path is internal:
* src/routePaths
* src/Containers/MainPage/Body.tsx
* src/Containers/Sidebar/NavigationSidebar.tsx

It is reasonable to suppose that integration and manual testing will quickly find any inconsistency.

During the initial attempt, I tried a TypeScript string union, but was not confident of benefit to cost.